### PR TITLE
#2288 Move nodeNames checking to creation/update phase for stream route

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2288)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala [hunk 1]
```diff
@@ -106,7 +107,7 @@ object StreamApi {
       // the default value
       .nullToString(StreamDefUtils.IMAGE_NAME_DEFINITION.key(), IMAGE_NAME_DEFAULT)
       .nullToRandomPort(StreamDefUtils.JMX_PORT_DEFINITION.key())
-      .nullToInt(StreamDefUtils.INSTANCES_DEFINITION.key(), 1)
+      //TODO remove this default value after #2288
```
Comment (chia7712) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala]: seems you have removed it in this PR?

Comment (saivirtue) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala]: this line change aim to:
1. remove default value for `instances`
2. add comment for `nodeNames` that it should not have default value

Comment (chia7712) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/StreamApi.scala]: Got it.

**Comment by saivirtue** [comment]
retry

**Comment by saivirtue** [comment]
retry configurator

**Pull request merged**